container-disc/src/main/java/com/yahoo/container/jdisc/component/Deconstructor.java

```diff
@@ -52,27 +54,38 @@ public void deconstruct(Object component) {
     }
 
     private static class DestructComponentTask implements Runnable {
+
         private final AbstractComponent component;
 
         DestructComponentTask(AbstractComponent component) {
             this.component = component;
         }
 
+        /** Returns a random value which will be different across identical containers invoking this at the same time */
+        private long random() {
+            return new SecureRandom().nextLong();
+        }
+
         public void run() {
             log.info("Starting deconstruction of " + component);
             try {
                 component.deconstruct();
                 log.info("Finished deconstructing " + component);
-            } catch (Error e) {
+            }
+            catch (Exception | NoClassDefFoundError e) { // May get class not found due to it being already unloaded
+                log.log(WARNING, "Exception thrown when deconstructing " + component, e);
+            }
+            catch (Error e) {
                 try {
-                    Thread.sleep((long) (new Random(System.nanoTime()).nextDouble() * 180 * 1000));
-                } catch (InterruptedException exception) { }
+                    // Randomize restart over 10 minutes to avoid simultaneous cluster restarts
+                    Thread.sleep(random() * 1000 * 60 * 10);
+                }
+                catch (InterruptedException exception) {
+                    log.log(WARNING, "Randomized wait before dying disrupted. Dying now.");
+                }
                 com.yahoo.protect.Process.logAndDie("Error when deconstructing " + component, e);
```

Comment:
Add shutting down or something to the log message. "Shutting down due to error when deconstruction .. "

Reply:
Done.